### DISCUSSION TITLE
Contributing guidelines | Project formating and linting | CI/CD

### DISCUSSION THREAD

**Discussion by xelacast** [FlowTestAI/FlowTest#94] [Ideas]
I have a PR draft #93 in progress and I ran into a few issues

## Formatting and Linting. 
The commands were not working so I made them target everything except node_modules and the build folders. There are ~86 errors in linting when I run pnpm lint and pnpm lint:fix won't fix them. We'll have to make sure the .eslintrc.js file is in good standing and then fix all the linting errors in the files. Those pesky errors can get annoying quickly but it will improve the workflow.

Enhancement: Adding a ```pnpm check``` command step to check for lint errors and formatting errors before the users make a PR

## Contributions
I added the CONTRIBUTING.md file to guide and streamline contributions.
I added [changeset](https://github.com/changesets/changesets) to the pnpm workspace for version control but there are a few concerns noted in the draft from CI/CD and *.md commit code bloat that would need to be taken care of.

## GitHub Actions for contributing
- We would need a build and build time check
- Linting check
- Formatting check
- Changeset integration to offload .md summary inserts  Look [here](https://www.npmjs.com/package/@changesets/changelog-github)
- Changeset version update 
- and more that I can't name off of the top of my head to keep up to standards

This is based on my current knowledge if anyone has more input please feel free to chat and/or give advice 💯

**Comment by jsajal** [comment]
Thank you for starting this discussion. We often forget the importance of a good development workflow in shaping a product.

> We would need a build and build time check

This is something that we need the most as of now. 

> Linting check and Formatting check

I will work on fixing current lint errors and after that this can be done too

> Changeset integration to offload .md summary inserts Look [here](https://www.npmjs.com/package/@changesets/changelog-github)
> Changeset version update

This is something that can wait, I think due to reasons noted here https://github.com/FlowTestAI/FlowTest/pull/93#pullrequestreview-2039170769

We are focused too much on core-features and user feedbacks right now so I will leave it to the community for more input and discussions but so far your approach sounds good.
